File: src/eSims.Data/Building/Room.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;
using eSims.Data.HumanResources;
using Newtonsoft.Json;

namespace eSims.Data.Building
{
  public class Room : RoomTemplate
  {
    public int RoomTemplateId { get; set; }

    //Position

    public int Left { get; set; }
    public int Top { get; set; }

    public int Rotation { get; set; }

    //Functions

    public int WorkplaceCount { get; set; }

    public int BathroomCount { get; set; }

    public int SmokeCount { get; set; }

  }
}

File: src/eSims/Client/scripts/building/room.ts
import { RoomTemplate } from './room.template';

export class Room extends RoomTemplate {
  public roomTemplateId: number;

  public left: number;
  public top: number;
  public rotation: number;

  public workplaceCount: number;
  public bathroomCount: number;
  public smokeCount: number;
  public kitchenCount: number;

}